packages/test/src/test-nexus-handler.ts

```diff
@@ -0,0 +1,524 @@
+import { randomUUID } from 'node:crypto';
+import anyTest, { TestFn } from 'ava';
+import getPort from 'get-port';
+import * as nexus from 'nexus-rpc';
+import * as protoJsonSerializer from 'proto3-json-serializer';
+import * as temporalnexus from '@temporalio/nexus';
+import * as root from '@temporalio/proto';
+import * as testing from '@temporalio/testing';
+import { DefaultLogger, LogEntry, Runtime, Worker } from '@temporalio/worker';
+import { ApplicationFailure, CancelledFailure, defaultFailureConverter, defaultPayloadConverter, SdkComponent } from '@temporalio/common';
+import { cleanStackTrace } from './helpers';
+
+export interface Context {
+  httpPort: number;
+  taskQueue: string;
+  endpointId: string;
+  env: testing.TestWorkflowEnvironment;
+  logEntries: LogEntry[];
+}
+
+const test = anyTest as TestFn<Context>;
+
+test.before(async (t) => {
+  const logEntries = new Array<LogEntry>();
+
+  const logger = new DefaultLogger('INFO', (entry) => {
+    logEntries.push(entry);
+  });
+
+  Runtime.install({ logger });
+  t.context.httpPort = await getPort();
```

Comment:
TODO(JWH): This will be flaky in CI due to port collisions, and users will often need to do that themselves. Move this to Core.

Reply:
Replaced by our existing `getRandomPort` helper.